pathy/__init__.py

```diff
@@ -241,36 +235,10 @@ def mkdir(self, path: "Pathy", mode: int = 0) -> None:
             self.create_bucket(path)
 
 
-class _PathyFlavour(_PosixFlavour):  # type:ignore
-    sep: str
-    is_supported = True
-
-    def parse_parts(self, parts: List[str]) -> Tuple[str, str, List[str]]:
-        parse_tuple: Tuple[str, str, List[str]] = super().parse_parts(  # type:ignore
-            parts
-        )
-        drv, root, parsed = parse_tuple
-        if len(parsed) and parsed[0].endswith(":"):
-            if len(parsed) < 2:
-                raise ValueError("need atleast two parts")
-            # Restore the
-            drv = parsed[0]  # scheme:
-            root = parsed[1]  # bucket_name
-        for part in parsed[1:]:
-            if part == "..":
-                index = parsed.index(part)
-                parsed.pop(index - 1)
-                parsed.remove(part)
-        return drv, root, parsed
-
-    def make_uri(self, path: "Pathy") -> str:
-        return str(path)
-
-
-class PurePathy(PurePath):
-    """PurePath subclass for bucket storage."""
+class PurePathy(PurePathBase):
+    """PurePathBase subclass for bucket storage."""
 
-    _flavour = _PathyFlavour()
+    pathmod = pathy_pathmod
```

Comment:
@barneygale this will be a problem, right?

Reply:
Can you tell me why you think that? I'm happy to fix something if it's wrong. 

This code is exercised a bunch by the test suite, and the behaviors are as expected across Python (3.8-3.12), so hopefully it's not _too wrong_.

I assumed that the introduction of pathmod here was to make it more easily accessible to the path classes (e.g., you can say `some_path.pathmod.sep` and get the proper separator to use.

Reply:
Oh, because this attribute is not public in pathlib anymore: https://github.com/python/cpython/issues/100502#issuecomment-1863447579

Reply:
I think I understand, thanks. 🙇 

I removed the assignment of `pathmod` with Pathy's specific module, but it caused cascading failures in my test suite. 

The trouble is that the `pathmod` defines how to split the string for root/drv/tail, and without it, my bucket paths get mangled. 

If we shouldn't override pathmod, we'll need another way to control the path-splitting logic.

Reply:
Because this is functional and fixes our broken main branch build, I will merge this now. 

Please feel free to comment here or in another issue and we can refine this implementation where needed.

Reply:
@merwok `pathmod` is still public! You were right to urge me to keep it, as it rather nicely abstracts away all the pure functionality of the path using composition rather than inheritance.

I'm thinking of adding a `PathModuleBase` to define its interface: #113893

Reply:
It is?  Thought it was added in 3.13 then removed (so renamed to `_pathmod` in the end)

Reply:
I considered renaming it to `_pathmod` but decided against it, given your advice to wait: https://github.com/python/cpython/issues/100502#issuecomment-1863541964